.github/workflows/ci-go.yml

```
@@ -0,0 +1,51 @@
+name: ci-go
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  go-fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: v1.16
+
+      - name: Check gofmt
+        run: gofmt -s -d $(find . -type f -name '*.go' -print)
```

```suggestion
        run: test -z $(gofmt -s -l $(find . -type f -name '*.go' -print) | tee /dev/stderr)
```
gofmt doesn't fail even if the files are not formatted right. This should help with realising failures.